src/utils/handlers/farmerHandlers.js

```diff
@@ -61,43 +60,49 @@ export const addFarmerHandler = (newFarmer, token) => {
   }
 
   return axios
-    .post(`${pathObj.addFarmerPath}`, setHeaders(token), newFarmer)
+    .post(`${pathObj.addFarmerPath}`, 
+      newFarmer,
+      setHeaders(), 
+    )
     .then(res => {
       if (res.data.successMessage) {
         return res.data;
       }
     })
     .catch(error => {
-      throw new Error(error);
+      // Note that this *should* not throw a new Error, but rather return the error array. 
+      // This gets used to generate the toasts on the form itself.
+      throw error;
     });
 };
 
-export const updateFarmerHandler = (changes, farmerId, token) => {
+export const updateFarmerHandler = (changes, farmerId) => {
   // Once again, add a separate function to exhaustively check all incoming inputs
   if (!changes) {
     throw new Error("Make sure you're passing valid changes!");
   }
 
-  return axios
-    .patch(
-      `${pathObj.updateFarmerPath}/${farmerId}/update`,
-      changes,
-      setHeaders(token)
-    )
+  return axios.patch(`${pathObj.updateFarmerPath}/${farmerId}/update`,
+    changes,
+    setHeaders(),
+  )
     .then(res => {
       if (res.data.success) {
         return res.data;
       }
     })
     .catch(error => {
+      // Note that this *should* not throw a new Error, but rather return the error array. 
```

Comment:
What do you mean by that, why shoud the error not be thrown?

Reply:
If it throws the error, no array of errors is returned to the component. The component uses this array of errors to then generate the respective toasts.